apps/src/p5lab/spritelab/SpritelabInput.jsx

```diff
@@ -59,9 +60,16 @@ class SpritelabInput extends React.Component {
 
   constructSpriteMap = memoize(animationPropsByKey => {
     const spriteMap = {};
-    Object.values(animationPropsByKey).forEach(
-      animation => (spriteMap[`image_${animation.name}`] = animation.sourceUrl)
-    );
+
+    Object.entries(animationPropsByKey).forEach(animation => {
+      const animationKey = animation[0];
+      const animationData = animation[1];
+      const url =
+        animationData.sourceUrl ||
+        // If custom animation, generate URL from animations API
+        animationSourceUrl(animationKey, animationData, this.props.channelId);
+      spriteMap[`image_${animation[1].name}`] = url;
```

Comment:
For consistency, since we've named `animation[1]` as `animationData`, I think we can use that here.

Reply:
ah great catch! totally forgot to actually use the variable I made haha!

Reply:
updated: https://github.com/code-dot-org/code-dot-org/pull/45636/commits/de2e752831832b88479b4ca3eb55337de03ec365